lib/vocabularies/discriminator/index.ts

```diff
@@ -21,13 +21,23 @@ const def: CodeKeywordDefinition = {
   error,
   code(cxt: KeywordCxt) {
     const {gen, data, schema, parentSchema, it} = cxt
-    const {oneOf} = parentSchema
+
+    const keyword = parentSchema.hasOwnProperty("oneOf")
+      ? "oneOf"
+      : parentSchema.hasOwnProperty("anyOf")
+      ? "anyOf"
+      : undefined
+
     if (!it.opts.discriminator) {
       throw new Error("discriminator: requires discriminator option")
     }
     const tagName = schema.propertyName
     if (typeof tagName != "string") throw new Error("discriminator: requires propertyName")
-    if (!oneOf) throw new Error("discriminator: requires oneOf keyword")
+    if (!keyword)
+      throw new Error(
+        "discriminator: is legal only when using one of the composite keywords oneOf, anyOf"
```

Comment:
```suggestion
        "discriminator: requires the oneOf or anyOf composite keyword"
```

I propose to simplify the error message